src/coreComponents/fieldSpecification/SourceFluxStatistics.hpp

```diff
@@ -0,0 +1,420 @@
+   */
+  template< typename LAMBDA >
+  void forAllRegionStatsWrappers( MeshLevel & meshLevel, string_view fluxName, LAMBDA && lambda );
+  /**
+   * @brief Apply a functor to all subregion WrappedStats (of the given region) that target a given flux.
+   * @param region    the region from which we want to execute the lambda for each of its sub-region.
+   * @param fluxName  the name of the flux from which we want the statistics.
+   * @param lambda    the functor that will be called for each WrappedStats. Takes in parameter the
+   *                  ElementSubRegionBase reference and the reference of its WrappedStats.
+   * @tparam LAMBDA   the type of lambda function to call in the function
+   * @note To be retrieved, the WrappedStats structs must be registered on the container during the
+   * registerDataOnMesh() call.
+   */
+  template< typename LAMBDA >
+  void forAllSubRegionStatsWrappers( ElementRegionBase & region, string_view fluxName, LAMBDA && lambda );
+
+  /**
+   * @brief Apply a functor to all WrappedStats of the given group that target a given flux (and
+   * potentially multiple SourceFluxStatsAggregator).
+   * The functor takes in parameter the reference to the currently processed WrappedStats.
+   * @note To be retrieved, the WrappedStats structs must be registered on the container during the
+   * registerDataOnMesh() call.
+   * @tparam LAMBDA   the type of lambda function to call in the function
+   * @param container the container from which we want the statistics.
+   * @param fluxName  the name of the flux from which we want the statistics.
+   * @param lambda    the functor that will be called for each WrappedStats. Takes in parameter the
+   *                  reference to the currently processed WrappedStats.
+   */
+  template< typename LAMBDA >
+  static void forAllFluxStatWrappers( Group & container, string_view fluxName, LAMBDA && lambda );
+
+  /**
+   * @return a string used to name the wrapper that is added to each region that is simulated by the solver.
+   * The string is unique within the region for the SourceFluxBoundaryCondition and the SourceFluxStatsAggregator.
+   * @param fluxName The name of the flux. For the mesh-level global stats, fluxSetWrapperString() can be used.
+   */
+  inline string getStatWrapperName( string_view fluxName ) const;
+
+
+  /**
+   * @brief View keys
+   */
+  struct viewKeyStruct
+  {
+    /// @return The key for setName
+    constexpr inline static string_view fluxNamesString() { return "fluxNames"; }
+    /// @return The key for statistics wrapper name that targets all region
+    constexpr inline static string_view allRegionWrapperString() { return "all_regions"; }
+    /// @return The key for statistics wrapper name that targets all fluxes of the set
+    constexpr inline static string_view fluxSetWrapperString() { return "flux_set"; }
+  };
+
+protected:
+
+  /**
+   * @copydoc Group::postProcessInput()
+   */
+  void postProcessInput() override;
```

Comment:
Can you move this `protected` function between the `public` and the `private`?
+ Are you sure that you need it to be `protected`?

Reply:
I set the class as `final` and removed the protected qualificator.